tests/Moq.Tests/MockFixture.cs

```diff
@@ -467,6 +468,61 @@ public void ConstructsClassWithNoDefaultConstructorAndNullValue()
 			Assert.Equal(26, mock.Object.IntValue);
 		}
 
+		[Fact]
+		public void ConstructsObjectsWithCtorLambda()
+		{
+			var mock = new Mock<ClassWithNoDefaultConstructor>(
+				() => new ClassWithNoDefaultConstructor("Hello", 26),
+				MockBehavior.Default);
+
+			Assert.Equal("Hello", mock.Object.StringValue);
+			Assert.Equal(26, mock.Object.IntValue);
+		}
+
+		[Fact]
+		public void ConstructsObjectsWithCtorLambda_MethodCall()
+		{
+			var stringValue = new StringBuilder("Hello");
+			var mock = new Mock<ClassWithNoDefaultConstructor>(
+				() => new ClassWithNoDefaultConstructor(stringValue.ToString(), 26),
+				MockBehavior.Default);
+
+			Assert.Equal("Hello", mock.Object.StringValue);
+			Assert.Equal(26, mock.Object.IntValue);
+		}
+
+		[Fact]
+		public void ConstructsObjectsWithCtorLambda_MemberAccess()
+		{
+			var mock = new Mock<ClassWithNoDefaultConstructor>(
+				() => new ClassWithNoDefaultConstructor(new FooOverrideEquals { Name = "Hello" }.Name, 26),
+				MockBehavior.Default);
+
+			Assert.Equal("Hello", mock.Object.StringValue);
+			Assert.Equal(26, mock.Object.IntValue);
+		}
+
+		[Fact]
+		public void ConstructsObjectsWithCtorLambda_ArrayAccess()
+		{
+			var mock = new Mock<ClassWithNoDefaultConstructor>(
+				() => new ClassWithNoDefaultConstructor(new[] { "Hello" }[0], 26),
+				MockBehavior.Default);
+
+			Assert.Equal("Hello", mock.Object.StringValue);
+			Assert.Equal(26, mock.Object.IntValue);
+		}
+
+		[Fact]
+		public void ConstructsObjectsWithCtorLambda_NoParameter()
+		{
+			var mock = new Mock<FooOverrideEquals>(
+				() => new FooOverrideEquals(),
```

Comment:
It's definitely good to have a test for this use case... but it makes me wonder whether Moq should throw an `ArgumentException` in this case, since the lambda is completely unnecessary and the code could be written more simply. (An analyzer would be better suited for this kind of thing but we still haven't got one.)

That's not a request, just a thought. What's your opinion?

Reply:
Don't necessarily aligned with you on this: if I were to write test on a type that likely could receive new parameters in the future (new DI parameters or simple parameters), I could decide to use the lambda-based syntax as an efficient way to get compiler errors if my tests have to be updated.

I believe it's a call of the test writer to use this syntax or the previous one.